### PR TITLE
Theme Showcase: Revert new theme preview modal

### PR DESCRIPTION
**Pull request by taipeicoder**
## Proposed Changes

This PR removes code that implements the new theme preview modal via the feature flag `themes/showcase-i4/details-and-preview`. Since we are bringing style selection in the theme detail page instead, we'll move the Tracks events there after the work in the theme detail page is done.

## Testing Instructions

* Head to the logged-in Theme Showcase `/themes/${site_slug}`. If using calypso.live, the flag `themes/showcase-i4/details-and-preview` is required.
* Ensure that the live demo button opens up the old theme preview modal, not the new one.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/themes/theme-preview.jsx [hunk 1]
```diff
@@ -1,39 +1,26 @@
...
 import { Button } from '@automattic/components';
-import { getDesignPreviewUrl } from '@automattic/design-picker';
-import { createHigherOrderComponent } from '@wordpress/compose';
 import { localize } from 'i18n-calypso';
-import page from 'page';
 import PropTypes from 'prop-types';
 import { Component } from 'react';
 import { connect } from 'react-redux';
 import QueryCanonicalTheme from 'calypso/components/data/query-canonical-theme';
-import PremiumGlobalStylesUpgradeModal from 'calypso/components/premium-global-styles-upgrade-modal';
```
Comment (taipeicoder) [inline client/my-sites/themes/theme-preview.jsx:11]: Should we remove this component from the code base as well? 🤔

File: client/my-sites/themes/theme-preview.jsx [hunk 2]
```diff
@@ -1,39 +1,26 @@
...
 import PulsingDot from 'calypso/components/pulsing-dot';
-import ThemePreviewModal from 'calypso/components/theme-preview-modal';
```
Comment (taipeicoder) [inline client/my-sites/themes/theme-preview.jsx:13]: Should we remove this component from the code base as well? 🤔 We might need it in the future.

File: client/my-sites/themes/theme-preview.jsx (hunk 1 above)
Comment (arthur791004) [reply client/my-sites/themes/theme-preview.jsx:11]: Do you mean `PremiumGlobalStylesUpgradeModal`? If so, we can remove it and add it back when we need it in the future 🙂

File: client/my-sites/themes/theme-preview.jsx (hunk 2 above)
Comment (arthur791004) [reply client/my-sites/themes/theme-preview.jsx:13]: Same

File: client/my-sites/themes/theme-preview.jsx (hunk 1 above)
Comment (taipeicoder) [reply client/my-sites/themes/theme-preview.jsx:11]: Yep! Both `PremiumGlobalStylesUpgradeModal` and `ThemePreviewModal` were created for this new preview modal. Now that we are reverting it, we can remove it from the code as well.

**Review by arthur791004 — COMMENTED** [review]
Tested and the modal becomes the old one 👍

However, there are some issues:
* Clicking on the disc, the preview modal doesn't display the correct style variation
* In the theme detail page, if you select any style variation and click on "Open the live demo", the selected style variation will reset to the default one, and the preview modal doesn't display the correct style variation.

It might not be a blocker for this PR, but we have to fix it in the future 🙂

Besides, I think the new modal is great even if we bring the style variation to the theme detail page, maybe we can still keep it 🤔

**Comment by taipeicoder** [comment]
> Tested and the modal becomes the old one 👍
> 
> However, there are some issues:
> 
> * Clicking on the disc, the preview modal doesn't display the correct style variation
> * In the theme detail page, if you select any style variation and click on "Open the live demo", the selected style variation will reset to the default one, and the preview modal doesn't display the correct style variation.
> 
> It might not be a blocker for this PR, but we have to fix it in the future 🙂
> 
> Besides, I think the new modal is great even if we bring the style variation to the theme detail page, maybe we can still keep it 🤔

I can see `<ThemePreviewModal/>` being more useful in the future once we have more customization options (like fonts). We can refer back to this PR when we want to bring it back 🙂

File: client/my-sites/themes/theme-preview.jsx (hunk 1 above)
Comment (taipeicoder) [reply client/my-sites/themes/theme-preview.jsx:11]: I found that `<PremiumGlobalStylesUpgradeModal />` is used by the onboarding design picker ([src](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L21)) as part of a previous refactor ([PR](https://github.com/Automattic/wp-calypso/pull/71850/files)) so we can't remove it 😅

So we'll just remove `<ThemePreviewModal/>` for now.

**Review by arthur791004 — APPROVED** [review]
👍

**Pull request merged**